### PR TITLE
Improve getting generator/parse cache in module factory

### PR DESCRIPTION
**Pull request by smelukov**
**What kind of change does this PR introduce?**

Refactoring to getting cache for generator/parser in module/factory

**Did you add tests for your changes?**

No

**Does this PR introduce a breaking change?**

No

**What needs to be documented once your changes are merged?**

Nothing

### CODE REVIEW

File: lib/NormalModuleFactory.js [hunk 1]
```diff
@@ -621,19 +628,20 @@ class NormalModuleFactory extends ModuleFactory {
...
-	getParser(type, parserOptions) {
-		let ident = type;
-		if (parserOptions) {
-			if (parserOptions.ident) {
-				ident = `${type}|${parserOptions.ident}`;
-			} else {
-				ident = JSON.stringify([type, parserOptions]);
-			}
+	getParser(type, parserOptions = EMPTY_RESOLVE_OPTIONS) {
+		if (!this.parserCache.has(type)) {
```
Comment (sokra) [inline lib/NormalModuleFactory.js]: I dislike the double `has` `get` cycle. That kind of duplicate work.

Could you change it to:

``` js
let cache = this.parserCache.get(type);
if (cache === undefined) {
  cache = new WeakMap();
  this.parserCache.set(type, cache);
}
```

Similar for all other cache accesses
```

**Comment by smelukov** [comment]
> avoid has -> get, set -> get

@sokra it was already done in https://github.com/webpack/webpack/pull/9984/commits/53764f5d36207c42cd7c31f3791f6cb0d0d51a57 ;)

**Comment by sokra** [comment]
> it was already done in [53764f5](https://github.com/webpack/webpack/commit/53764f5d36207c42cd7c31f3791f6cb0d0d51a57) ;)

not all of it

**Comment by sokra** [comment]
Thanks

**Pull request merged**